sdk/cognitivelanguage/Azure.AI.Language.Conversations/src/ConversationAnalysisClient.cs

```diff
@@ -150,5 +153,136 @@ public virtual Response<AnalyzeConversationTaskResult> AnalyzeConversation(strin
+                var jobId = getJobId(responseHeaders.Headers.OperationLocation);
+
+                await WaitForJobUntilDoneAsync(jobId).ConfigureAwait(false);
+
+                return await _analysisRestClient.JobStatusAsync(jobId, cancellationToken: cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
+        }
+
+        private async Task<bool> WaitForJobUntilDoneAsync(Guid jobId, int timeoutInSeconds = 0, int pollingIntervalInSeconds = 5)
+        {
+            var startTimeStamp = DateTime.Now;
+            while (true)
+            {
+                var jobStatus = await _analysisRestClient.JobStatusAsync(jobId).ConfigureAwait(false);
+
+                // check job status is done
+                if (jobStatus.Value.Status == JobState.Succeeded || jobStatus.Value.Status == JobState.Failed || jobStatus.Value.Status == JobState.PartiallySucceeded)
+                    return true;
+
+                // check for timeouts
+                if (timeoutInSeconds > 0)
+                {
+                    var nowTimeStamp = DateTime.Now;
+                    var diff = nowTimeStamp.Subtract(startTimeStamp).TotalSeconds;
+                    if (diff > timeoutInSeconds)
+                    {
+                        break;
+                    }
+                }
+
+                // sleep for a while
+                Thread.Sleep(pollingIntervalInSeconds);
+            }
+            return false;
+        }
+
+        /// <summary>Analyzes a conversational utterance.</summary>
+        /// <param name="input">The <see cref="GeneratedConversation"/> used for tasks input.</param>
+        /// <param name="tasks"> <see cref="AnalyzeConversationLROTask"/> defines the tasks to be used.</param>
+        /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> to cancel the request.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="input"/> or <paramref name="tasks"/> or is null.</exception>
+        /// <exception cref="RequestFailedException">The service returned an error. The exception contains details of the service error.</exception>
+        public virtual Response<AnalyzeConversationJobState> AnalyzeConversation(IEnumerable<GeneratedConversation> input, IEnumerable<AnalyzeConversationLROTask> tasks, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNullOrEmpty(input, nameof(input));
+            Argument.AssertNotNull(tasks, nameof(tasks));
+
+            MultiLanguageConversationAnalysisInput multiLanguageConversationAnalysisInput = new MultiLanguageConversationAnalysisInput(input);
+
+            var analyzeConversationJobsInput = new AnalyzeConversationJobsInput(multiLanguageConversationAnalysisInput, tasks);
```

Comment:
Nit: be consistent with `var`. Visual Studio should warn you about our usage i.e. that you can use it (don't have to) for obvious instantiations, but should not use it elsewhere like below. You can `Ctrl+.` on the gray squiggles to autocorrect it.